backend/app/views/spree/admin/shared/_address_form.html.erb

```diff
@@ -1,42 +1,49 @@
 <% s_or_b = type.chars.first %>
 
 <div id="<%= type %>" data-hook="address_fields">
-  <div class="form-group <%= "#{type}-row" %>">
+  <%= field_container f.object, :firstname, class: ["form-group", "#{type}-row"] do %>
     <%= f.label :firstname, Spree.t(:first_name) %>
     <%= f.text_field :firstname, class: 'form-control' %>
-  </div>
+    <%= error_message_on f.object, :firstname %>
```

Comment:
@priyank-gupta in this particular form it won't work, that's why we need to explicitly pass `f.object` here